src/checkoutservice/genproto/oteldemo/demo.pb.go

```diff
@@ -0,0 +1,2739 @@
+// Copyright 2020 Google LLC
```

Comment:
the proto files should not be commited.
They are built within the dockerfile:
https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/checkoutservice/Dockerfile#L23

Reply:
The pb.go files were checked in to resolve the errors from Dependabot:


go: github.com/open-telemetry/opentelemetry-demo/src/checkoutservice imports
	github.com/open-telemetry/opentelemetry-demo/src/checkoutservice/genproto/oteldemo: cannot find module providing package github.com/open-telemetry/opentelemetry-demo/src/checkoutservice/genproto/oteldemo

Reply:
I think we should find another way of solving the dependabot issue.
The main idea is to not have to touch this file if the `pb` changes.

Reply:
@MrAlias @dashpole any ideas on a workaround for using dependabot without checking in pb.go files?

Reply:
Hmm, I'm not sure how to make this work without checking in all the Go source files that the package depends on. There is no GitHub action that I know you can hook into prior to the dependabot one runs.

FWIW, we commit the generated proto Go code in other repositories when it is used:

- https://github.com/open-telemetry/opentelemetry-proto-go/tree/main/otlp
- https://github.com/open-telemetry/opentelemetry-collector/tree/main/pdata/internal/data/protogen

And so do other projects:

- https://github.com/grpc/grpc-go/tree/master/examples/helloworld/helloworld
- https://github.com/grpc/grpc-go/blob/master/examples/route_guide/routeguide/route_guide_grpc.pb.go

It is a pretty standard practice to have all the Go code, generated or not, that comprises a package checked in to the used VCS.

Reply:
@MrAlias and how do you ensure the generated proto files committed are up to date?

Reply:
Our struggle here is the demo consists of several services across several languages, and they all are dependent on the same protobuf definition with the runtime or compile time generation of the code. We do this for maintenance reasons. Changing this just for Go services would require us to change how we manage our protobuf definition just for 1 language but not others.

Reply:
@MrAlias and @CharlieTLe can't the proto generation step be added to the dependabot action before we call the new action?

Reply:
@julianocosta89 can you link to the dependabot action definition? I thought that was completely automated?

Reply:
Seems like there is an open issue for this: https://github.com/dependabot/dependabot-core/issues/9638

Reply:
🤔 maybe I'm totally lost here, but isn't that exactly what we are updating in this PR?
`.github/dependabot.yml` would be the file.
Before running the dependabot action, can't we call another action?

Reply:
no, it cant execute workflows. I agree with OP that committing generated code is a good path forward and actually makes change explicit.

Reply:
If we do this, we need to commit generated code for all languages/services. I don't want to have special cases for specific languages or services in the demo.